tsconfig.json

```diff
@@ -0,0 +1,33 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "inlineSources": true,
+    "isolatedModules": true,
+    "jsx": "react",
+    "lib": ["dom", "es2020"],
+    "moduleResolution": "node",
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "outDir": "tsout",
```

Comment:
This setting will not end up getting used by the build system — [`tsify`](https://github.com/TypeStrong/tsify) will output JavaScript in-memory and then funnel the result through the build pipeline. This is only here so that if you want to test things out you can say `yarn tsc --project .` like [this](https://github.com/MetaMask/controllers/blob/81bcbae1c4abe71a9b07819606e155a33ec4a6b4/package.json#L32).